app/src/main/java/com/example/toyproject/ui/myskin/MySkinFragment.kt

```diff
@@ -0,0 +1,137 @@
+        setupDeleteItem()
+        setupSearch()
+        setUpChips()
+    }
+
+    private fun setupUI() {
+        setAdapter(binding.rvMySkin, adapter)
+        binding.searchView.setQuery("", false)
+    }
+
+    private fun setupDeleteItem() {
+        binding.imgDeleteAll.setOnClickListener {
+            val alert = AlertView("Delete Item All", "choice", AlertStyle.IOS)
+            alert.addAction(AlertAction("Delete All", AlertActionStyle.NEGATIVE) {
+                viewModel.deleteAllSkins()
+            })
+            alert.show(requireActivity() as AppCompatActivity)
+        }
+    }
+
+    private fun setupItems() {
+        viewModel.skinLiveData.observe(viewLifecycleOwner, Observer {
+            when (it) {
+                is Resource.Success -> {
+                    showProgress(false)
+                    showRecyclerView(true)
+                    adapter.submitList(it.data)
+                }
+                is Resource.Error -> {
+                    showRecyclerView(false)
+                    binding.txtNoItem.isVisible = true
+                }
+                Resource.Loading -> {
+                    showProgress(true)
+                    showRecyclerView(false)
+                }
+            }
+        })
+        viewModel.loadAllMySkin()
+    }
+
+    private fun setupSearch() {
+        binding.searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String?): Boolean {
+                return true
+            }
+
+            override fun onQueryTextChange(newText: String?): Boolean {
+                newText?.let { viewModel.queryChannel.offer(it) }
+                return true
+            }
+        })
+
+        viewModel.searchResult.observe(viewLifecycleOwner, Observer {
+            adapter.submitList(it)
+        })
+    }
+
+    private fun setUpChips() {
+        with(binding) {
+            chipToxin.onClick("toxin")
+            chipAcne.onClick("acne")
+            chipFiller.onClick("filler")
+            chipLifting.onClick("lifting")
+            chipWaxing.onClick("waxing")
+            chipInjection.onClick("inject")
+        }
+    }
+
+    private fun Chip.onClick(skinTitle: String) {
+        setOnCheckedChangeListener { _, isChecked ->
+            if (isChecked) {
+                viewModel.fetchBySkinTitle(skinTitle)
+            } else viewModel.loadAllMySkin()
+        }
+    }
+
+    private fun showRecyclerView(status: Boolean) {
+        binding.rvMySkin.isVisible = status
+    }
+
+    private fun showProgress(status: Boolean) {
+        binding.loadingView.isVisible = status
+    }
+
+    override fun onClick(skinKinds: String) {
+        val alert = AlertView("Item Delete One", "choice", AlertStyle.IOS)
+        alert.addAction(AlertAction("Delete", AlertActionStyle.NEGATIVE) {
+            viewModel.deleteOneSkin(skinKinds)
+        })
+        alert.show(requireActivity() as AppCompatActivity)
+    }
```

Comment:
init : `private lateinit var alert: AlertView`

```suggestion
       if(::alert.isInitialized.not()) {
            alert = AlertView("Item Delete One", "choice", AlertStyle.IOS)
        }
        alert.addAction(AlertAction("Delete", AlertActionStyle.NEGATIVE) {
            viewModel.deleteOneSkin(skinKinds)
        })
        alert.show(requireActivity() as AppCompatActivity)
    }
```

코드스타일 차이인 것 같은데, 요렇게도 가능해서 제안드려봅니다 : )

Reply:
이렇게도 할 수 있군요! 참고하겠습니다.